base/src/numbers.rs

```diff
@@ -123,10 +123,16 @@ impl BlockTime {
 #[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Hash, Default)]
 pub struct BlockTimeSpan(pub u64);
 
-// Consider switch to [ark-ff](https://docs.rs/ark-ff/0.3.0/ark_ff/biginteger/struct.BigInteger256.html)
 #[derive(Clone, Serialize, Deserialize, Default, PartialEq, Debug)]
 pub struct BigInt256(pub [u8; 32]);
 
+impl From<BigInt256> for ark_ff::BigInteger256 {
+    fn from(i: BigInt256) -> Self {
+        use ark_ff::bytes::FromBytes;
+        Self::read(&i.0[..]).unwrap()
```

Comment:
I think implementing Deref it will expose internal details (if we want to hide it in future) and was only intended to implement for smart pointer types when it got introduced. More details: https://stackoverflow.com/questions/45086595/is-it-considered-a-bad-practice-to-implement-deref-for-newtypes

Again it depends if we not want to expose the internal impl of the newtype.